ports/views.py

```diff
@@ -67,13 +67,17 @@ def variantlist(request, variant):
 # Views for port-detail page START
 def portdetail(request, name):
     try:
+        req_port = Port.objects.get(name__iexact=name)
+
+        all_latest_builds = BuildHistory.objects.all().order_by('port_name', 'builder_name', '-build_id').distinct('port_name', 'builder_name')
```

Comment:
I didn't investigate yet, but the builds are no longer displayed correctly. If you look at gnuplot, most of the builds are missing in the port health status.

Reply:
Yes, I am aware of the issue. This is a very peculiar problem. I have tried two ways I could think of doing this and both gave same incorrect results.

The version deployed on EC2 does not contain these commits and I see the exact same results there also. However, it was working fine  last I checked.

Reply:
Yes, I am aware of the issue. This is a very peculiar problem. I have tried two ways I could think of doing this and both gave same incorrect results.

The version deployed on EC2 does not contain these commits and I see the exact same results there also. However, it was working fine  last I checked.